python/triton/compiler/compiler.py

```diff
@@ -246,7 +246,8 @@ def compile(src, target=None, options=None):
     enable_ir_dump = os.environ.get("TRITON_KERNEL_DUMP", "0") == "1"
     fn_override_manager = get_override_manager(src.hash()) if enable_override else None
     fn_dump_manager = get_dump_manager(src.hash()) if enable_ir_dump else None
-    metadata_filename = f"{src.name}.json"
+    file_name = src.name[:50]
```

Comment:
Could you please justify the choice of `50` a bit? Will it still be unique among all files if we take the first 50 characters?

Reply:
50 came out of our use case, which makes the eventual path length closer to 255.

I didn't realize file names need to be unique. Is it a hard requirement? I can use name hash instead.

Reply:
See

```
metadata_group[metadata_filename] = fn_cache_manager.put(json.dumps(metadata, default=vars), metadata_filename,
```

Reply:
`fn_cache_manager` is already unique to each kernel right? 

    fn_cache_manager = get_cache_manager(hash)

File name doesn't seem to be necessarily unique since the direct path (named based on `hash`) is unique. I was thinking if the runtime requires unique .cubin files. It shouldn't right, as with autotune we could run same-named kernels multiple times?

Reply:
Autotuner is applied per `JITFunction` so won't be affected

Reply:
I got your point now. Maybe it makes more sense to calculate a hash instead of truncating. cc @bertmaher What do you think?

Reply:
I'd lean towards truncation since it's occasionally nice to see the kernel names in the filename, and I don't think hashing this component is necessary for uniqueness.

We could probably use a bit more than 50 characters, since the constraint being hit is the 255 character `NAME_MAX` limit, and we're limited by the length of `{filepath}.tmp.pid_{pid}_{uuid.uuid4()}`.  In any case I'd probably put a comment here to explain why we're truncating